cmd/driver-registrar/main.go

```diff
@@ -50,14 +56,122 @@ const (
 
 // Command line flags
 var (
-	kubeconfig        = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
-	connectionTimeout = flag.Duration("connection-timeout", 1*time.Minute, "Timeout for waiting for CSI driver socket.")
-	csiAddress        = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
-	showVersion       = flag.Bool("version", false, "Show version.")
-
-	version = "unknown"
+	kubeconfig              = flag.String("kubeconfig", "", "Absolute path to the kubeconfig file. Required only when running out of cluster.")
+	connectionTimeout       = flag.Duration("connection-timeout", 1*time.Minute, "Timeout for waiting for CSI driver socket.")
+	csiAddress              = flag.String("csi-address", "/run/csi/socket", "Address of the CSI driver socket.")
+	kubeletRegistrationPath = flag.String("kubelet-registration-path", "/var/lib/kubelet/plugins/csi-hostpath/csi.sock", "Full path to CSI driver's socket.")
```

Comment:
Let's make the description even more descriptive. How about: `Enables Kubelet Plugin Registration service, and returns the specified path as "endpoint" in "PluginInfo" response. If this option is set, the driver-registrar expose a unix domain socket to handle Kubelet Plugin Registration, this socket MUST be surfaced on the host in the kubelet plugin registration directory (in addition to the CSI driver socket). If plugin registration is enabled on kubelet (kubelet flag KubeletPluginsWatcher is set), then this option should be set and the value should be the path of the CSI driver socket on the host machine.`

Reply:
Would it not make more sense to have the detailed description in the doc? I was planning to push a doc update once the new registrar is merged. WDYT?

Reply:
I suggest doing both. It's pretty ugly but I want to make sure people understand how this works.